src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -1689,6 +1691,234 @@ class spell_dk_death_grip_initial : public SpellScriptLoader
     }
 };
 
+enum RaiseAllyMisc
+{
+    TEXT_RISE_ALLY = 33055,
+
+    SPELL_DK_RISEN_GHOUL_SELF_STUN = 47466,
+    SPELL_DK_RISEN_GHOUL_SPAWN__IN = 47448,
+    SPELL_DK_SUMMON_HEAL = 36492,
+    SPELL_DK_DEATH_KNIGHT_RUNE_WEAPON_SCALING_02 = 51906,
+    SPELL_DK_DEATH_KNIGHT_PET_SCALING_01 = 54566,
+    SPELL_DK_DEATH_KNIGHT_PET_SCALING_03 = 61697,
+    SPELL_DK_MIRROR_NAME = 62224,
+    SPELL_DK_PET_SCALING___MASTER_SPELL_03___INTELLECT_SPIRIT_RESILIENCE = 67557,
+    SPELL_DK_PET_SCALING___MASTER_SPELL_06___SPELL_HIT_EXPERTISE_SPELL_PENETRATION = 67561,
+
+    SPELL_GHOUL_FRENZY = 62218,
+
+    DATA_CHARMER_GUID = 1,
+
+    NPC_RISEN_ALLY = 30230
+};
+
+// 61999 - Raise Ally Initial
+class spell_dk_raise_ally_initial : public SpellScriptLoader
+{
+public:
+    spell_dk_raise_ally_initial() : SpellScriptLoader("spell_dk_raise_ally_initial") { }
+
+    class spell_dk_raise_ally_initial_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_dk_raise_ally_initial_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_DK_RAISE_ALLY_INITIAL))
+                return false;
+            return true;
+        }
+
+        SpellCastResult CheckCast()
```

Comment:
I think you should check cast for player targets and also spell shoudn't be used in arena. Try this:
``cpp
            SpellCastResult CheckTarget()
            {
                if (Unit* target = GetTargetUnit())
                {
                    if (target->GetTypeId() != TYPEID_PLAYER || target->isAlive())
                        return SPELL_FAILED_BAD_TARGETS;
                    else if (GetCaster()->ToPlayer()->InArena())
                        return SPELL_FAILED_NOT_IN_ARENA;

                    return SPELL_CAST_OK;
                }
            }
``